src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -190,6 +193,8 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
             $this->boot();
         }
 
+        $request->setIpRetriever($this->container->get('ip_retriever'));
```

Comment:
This would be more the job of an event listener to inject the service into the request when this is necessary.